x/auth/ante.go

```diff
@@ -234,6 +235,11 @@ func consumeSignatureVerificationGas(meter sdk.GasMeter, pubkey crypto.PubKey, p
 		meter.ConsumeGas(params.SigVerifyCostED25519, "ante verify: ed25519")
 	case strings.Contains(pubkeyType, "secp256k1"):
 		meter.ConsumeGas(params.SigVerifyCostSecp256k1, "ante verify: secp256k1")
+	case strings.Contains(pubkeyType, "multisigthreshold"):
+		multisigPubKey := pubkey.(multisig.PubKeyMultisigThreshold)
+		for _, pk := range multisigPubKey.PubKeys {
```

Comment:
Should this consume gas based on the number of signatures in the transaction or based on the number of pubkeys in the threshold?

mostly I think it should be number of signatures but possibly could be persuaded it should be the number of pubkeys.

Reply:
I agree @zaki, it should be based on signatures. Will amend accordingly.

Reply:
I think there's an argument for `x * num pubkeys + y * num signatures`, since we have to do some processing for both (unmarshaling the public keys) - but `x` will be small compared to `y`.

Reply:
@cwgoes I've addressed @zmanian's comment - please have a look at the latest commit.